// @dc/eslint-config/es6.js
module.exports = {
  extends: 'airbnb',
  env: {
    browser: true
  },
  rules: {
    'arrow-body-style': ['error', 'as-needed'],
    'class-methods-use-this': 'off',
    'comma-dangle': ['error', 'never'],
    'func-names': ['error', 'as-needed'],
    'func-style': ['error', 'declaration', { allowArrowFunctions: true }],
    'function-paren-newline': ['error', 'consistent'],
    'max-len': ['error', { code: 100, ignoreTrailingComments: true, ignoreUrls: true }],
    'no-bitwise': 'off',
    'no-param-reassign': ['warn'],
    'no-plusplus': 'off',
    'no-prototype-builtins': 'off',
    'no-restricted-syntax': ['error', 'LabeledStatement', 'WithStatement'],
    'no-trailing-spaces': ['error', { skipBlankLines: true }],
    'no-unneeded-ternary': ['error', { defaultAssignment: true }],
    'no-unused-vars': ['error', { vars: 'local', args: 'none' }],
    'no-use-before-define': ['error', 'nofunc'],
    'object-shorthand': ['warn', 'consistent-as-needed'],
    'operator-linebreak': ['error', 'after', { overrides: { '?': 'before', ':': 'before' } }],
    'padded-blocks': ['off', 'never'],
    'prefer-destructuring': ['error', {
      AssignmentExpression: {
        array: false,
        object: false
      }
    }],
    'quote-props': ['error', 'consistent-as-needed'],
    'space-before-function-paren': ['error', { anonymous: 'never', named: 'never' }],
    'vars-on-top': 'warn',

    'import/extensions': ['off', 'always'],
    'import/prefer-default-export': 'off',

    'jsx-a11y/anchor-is-valid': 'warn',
    'jsx-a11y/click-events-have-key-events': 'warn',
    'jsx-a11y/label-has-for': 'warn',
    'jsx-a11y/mouse-events-have-key-events': 'warn',
    'jsx-a11y/no-autofocus': 'warn',
    'jsx-a11y/no-noninteractive-element-interactions': 'warn',
    'jsx-a11y/no-noninteractive-tabindex': 'warn',
    'jsx-a11y/no-static-element-interactions': 'warn',

    'react/jsx-filename-extension': ['error', { extensions: ['.js', '.jsx'] }],
    'react/no-array-index-key': 'warn',
    'react/no-unused-prop-types': 'off',
    'react/no-unused-state': 'warn',
    'react/style-prop-object': 'off'
  }
};
